### PR TITLE
Add demo overview doc

### PR DESCRIPTION
**Pull request by Aymalla**
Add a demo overview document that will be used as a presentation.

### CODE REVIEW

File: data-reconciliation-app/demo-app/demo-overview.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+In secure multi-party computing, encrypted data goes into the enclave. The enclave decrypts the data using a key, performs analysis, gets a result, and sends back an encrypted result that a party can decrypt with the designated key.
+
+[Secure multi-party systems](https://learn.microsoft.com/en-us/azure/confidential-computing/use-cases-scenarios) can be developed using CCF framework which could be written in [TypeScript, JavaScript, C++, and more will be supported in the future]
+
+![Applications](https://learn.microsoft.com/en-us/azure/confidential-computing/media/use-cases-scenarios/use-cases.png)
+
+
+## Data Reconciliation Application
+
+**Use Case:** Allow multi parties (organisations - Banks - companies) to process their confidential data **without exposing the input data to other parties**. This type of secure computation enables scenarios such as anti-money laundering, fraud-detection, and secure analysis of healthcare data.
```
Comment (ross-p-smith) [inline data-reconciliation-app/demo-app/demo-overview.md:47]:
```suggestion
**Use Case:** Allow multiple parties (organisations - Banks - companies) to process their confidential data **without exposing the input data to other parties**. This type of secure computation enables scenarios such as anti-money laundering, fraud-detection, and secure analysis of healthcare data. 
```

File: data-reconciliation-app/demo-app/demo-overview.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+**Data Schema:** Data schema will be agreed by members 
+- Unique key: uniquely identifying each record.
+- Value: to be reconciled with other members' data. 
+
+![Input Data](../docs/images/data.png)
+
+![Output Report](../docs/images/report.png)
+
+## Functional  Requirements
+- Members can submit their data as single or batch records
```
Comment (ross-p-smith) [inline data-reconciliation-app/demo-app/demo-overview.md:62]:
```suggestion
- Members can submit their data as a single record or a batch
```

File: data-reconciliation-app/demo-app/demo-guidance.md [hunk 3]
```diff
@@ -15,12 +15,12 @@ We have sample sets [here](./data/) for the FSI Demo. Brent will provide real da
 1. Deploy data reconciliation
```
Comment (ross-p-smith) [inline data-reconciliation-app/demo-app/demo-guidance.md:15]: Might play havoc with your links - but do you think that this document would be better off in the data-reconciliation-app/docs folder. Then we also slowly delete this demo folder?

Comment (Aymalla) [reply data-reconciliation-app/demo-app/demo-guidance.md:15]: Agree, but we can do it as seperate PR

**Review by ross-p-smith — APPROVED** [review]

**Pull request merged**